chain/ethereum/src/adapter.rs

```diff
@@ -303,24 +303,31 @@ impl EthereumCallFilter {
             return false;
         }
 
+        let v = self.contract_addresses_function_signatures.get(&call.to);
+
         // Ensure the call is to a contract the filter expressed an interest in
-        match self.contract_addresses_function_signatures.get(&call.to) {
-            None => false,
-            Some(v) => {
-                let signature = &v.1;
-
-                // If the call is to a contract with no specified functions, keep the call
-                //
-                // Allows the ability to genericly match on all calls to a contract.
-                // Caveat is this catch all clause limits you from matching with a specific call
-                // on the same address
-                if signature.is_empty() {
-                    true
-                } else {
-                    // Ensure the call is to run a function the filter expressed an interest in
-                    signature.contains(&call.input.0[..4])
-                }
-            }
+        if let None = v {
+            return false;
+        }
+
+        let signature = &v.unwrap().1;
```

Comment:
```suggestion
        let signature = match v {
          None => return false,
          Some(v) => &v.1,
        }
```

Reply:
I've wrapped this suggestion with the `self.contract_addresses_function_signatures.get(&call.to)` function call, so there's no more `v` let definition.